deps/sketch.py

```diff
@@ -0,0 +1,139 @@
+"""Sketch and library management"""
+
+import asyncio
+import base64
+import os
+import threading
+from contextlib import asynccontextmanager
+from math import floor
+
+import aiofiles
+from fastapi import FastAPI, HTTPException
+
+from conf import settings
+from deps.logs import logger
+from models import Library, Sketch
+
+CWD = os.getcwd()
+
+fqbn_to_board = {  # Mapping from fqbn to PlatformIO board
+    "arduino:avr:uno": "uno",
+    "arduino:avr:nano": "nanoatmega328",
+    "arduino:avr:mega": "megaADK",
+    "arduino:esp32:nano_nora": "arduino_nano_esp32",
+    "arduino:mbed_nano:nanorp2040connect": "nanorp2040connect",
+}
+
+fqbn_to_platform = {  # Mapping from fqbn to PlatformIO platform
+    "arduino:avr:uno": "atmelavr",
+    "arduino:avr:nano": "atmelavr",
+    "arduino:avr:mega": "atmelavr",
+    "arduino:esp32:nano_nora": "espressif32",
+    "arduino:mbed_nano:nanorp2040connect": "raspberrypi",
+}
+
+
+async def install_libraries(  # pylint: disable=too-many-locals, too-many-branches
```

Comment:
We used to add installed libs to a cache so we don't invoke IO heavy stuff on every compile. I think this is still needed, install_libs is called for every compile and we have a _lot_ of libs.